manticore/core/cpu/abstractcpu.py

```diff
@@ -552,6 +552,25 @@ def read_bytes(self, where, size):
             result.append(Operators.CHR(self.read_int(where + i, 8)))
         return result
 
+    def write_string(self, where, string, max_length=None):
+        '''
+        Writes a str to memory, appending a NULL-terminator at the end.
+        :param int where: Address to write the string to
+        :param int max_length:
+            The size in bytes to cap the string at, or None [default] for no
+            limit. This includes the NULL terminator.
+        :type string: str
+        '''
+        
+        if max_length is not None:
+            string = string[:max_length-1]
+        
+        # Conver the string to a list, and append 0 to it.
```

Comment:
Two comments, one stylistic and one structural: we generally prefer not to comment _what_ the code does, but _how_ it does it, so commenting that you're converting and appending is probably not needed. Second, this feels verbose and since `write_string` is a function that might see regular use in the future, should not perform unneeded conversions. I'd be inclined to replace following three lines with just:

`self.write_bytes(string + chr(0))` (or `'\x00'` instead of the cast)